sympy/logic/boolalg.py

```diff
@@ -1304,6 +1304,87 @@ def append_symbol(arg, symbols):
             for c in clauses]
 
 
+def term_to_integer(term):
+	"""
+	Constructs an integer from its base 2 digits.
+
+	Examples
+	========
+
+	>>> from sympy.logic.boolalg import term_to_integer
+	>>> term_to_integer([1, 0, 1, 1])
+	11
+	>>> terms = [[0, 0, 0, 1], [0, 0, 1, 1],
+	...			[0, 1, 1, 1], [1, 0, 1, 1], [1, 1, 1, 1]]
+	>>> map(term_to_integer, terms)
+	[1, 3, 7, 11, 15]
+
+	"""
+
+	term = list(term)
+	return int(''.join(map(str, term)), 2)
+
+
+def integer_to_term(i, n_bits=0):
+	"""
+	Gives a list of the base 2 digits in the integer i.
+	If additional n_bits argument is passed,
+	integer_to_term() extends list of digits with zeros.
+
+	Examples
+	========
+
+	>>> from sympy.logic.boolalg import integer_to_term
+	>>> integer_to_term(14)
+	[1, 1, 1, 0]
+	>>> integer_to_term(7, 5)
+	[0, 0, 1, 1, 1]
+	>>> truthtable = [1, 0, 0, 1, 1, 1, 0, 1]
+	>>> minterms = []
+	>>> n_bits = len(truthtable).bit_length() - 1
+	>>> for i, value in enumerate(truthtable):
+	...			if value:
+	...				minterms.append(integer_to_term(i, n_bits))
+	>>> minterms
+	[[0, 0, 0], [0, 1, 1], [1, 0, 0], [1, 0, 1], [1, 1, 1]]
+
+	"""
+	n = i.bit_length() if i else 1
+	prefix = [0] * (n_bits - n if n_bits else 0)
+	return prefix + map(int, list('{0:b}'.format(i)))
+
+
+def truth_table(expr, variables):
+	"""
+	Gives a list of truth values of the boolean expression.
```

Comment:
Thanks for your suggestions.

What do you think about using a generator as an output for <code>truth_table</code>?:

	def truth_table(expr, variables, input=True):
		"""
		Gives a generator of all possible configurations of the input variables,
		and the result of the boolean expression for those values.

		Parameters
		==========

		expr : string or boolean expression
		variables : list of variables
		input : boolean (default True)
			indicates whether to return the input combinations.

		Examples
		========
		>>> from sympy.logic.boolalg import truth_table
		>>> from sympy.abc import x,y
		>>> table = truth_table(x >> y, [x, y])
		>>> for t in table:
		>>>     print t[0], '->', t[1]
		[0, 0] -> True
		[0, 1] -> True
		[1, 0] -> False
		[1, 1] -> True

		>>> table = truth_table('x | y', ['x', 'y'])
		>>> print list(table)
		[([0, 0], False), ([0, 1], True), ([1, 0], True), ([1, 1], True)]

		If input is false, truth_table returns only a list of truth values.
		In this case, the corresponding input values of variables can be
		deduced from the index of a given output.

		>>> from sympy.logic.boolalg import integer_to_term
		>>> vars = [y, x]
		>>> values = truth_table(x >> y, vars, input=False)
		>>> values = list(values)
		>>> print values
		[True, False, True, True]

		>>> for i, value in enumerate(values):
		...     print zip(vars, integer_to_term(i, len(vars))), '->', value
		[(y, 0), (x, 0)] -> True
		[(y, 0), (x, 1)] -> False
		[(y, 1), (x, 0)] -> True
		[(y, 1), (x, 1)] -> True

		"""
		variables = [sympify(v) for v in variables]

		expr = sympify(expr)
		if not isinstance(expr, BooleanFunction):
			yield expr

		table = product([0, 1], repeat=len(variables))
		for term in table:
			term = list(term)
			value = expr.xreplace(dict(zip(variables, term)))

			if input:
				yield term, value
			else:
				yield value


Truth tables have the exponential growth in size as the number of variables increase. It would be better to iterate through truth table chunk by chunk, if the number of variables is large.